tests/test_fastapi_roles.py

```diff
@@ -0,0 +1,209 @@
+from typing import Awaitable, Callable, Tuple
+
+import pytest
+from fastapi import FastAPI, HTTPException, Request, Response
+from fastapi.testclient import TestClient
+
+# For tracing test
+from opentelemetry.sdk import trace as tracesdk
+
+# from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import SimpleSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+
+# Public API
+from dbos import DBOS, DBOSContextEnsure
+
+# Private API because this is a unit test
+from dbos.context import assert_current_dbos_context
+from dbos.error import DBOSDuplicateWorkflowEventError, DBOSNotAuthorizedError
+from dbos.system_database import GetWorkflowsInput
+from dbos.tracer import dbos_tracer
+
+
+@pytest.mark.order(1)
+def test_simple_endpoint(dbos_fastapi: Tuple[DBOS, FastAPI]) -> None:
+    # Set up a simple in-memory span exporter for testing
+    exporter = InMemorySpanExporter()
+    span_processor = SimpleSpanProcessor(exporter)
+    provider = tracesdk.TracerProvider()
+    provider.add_span_processor(span_processor)
+    dbos_tracer.set_provider(provider)
+
+    dbos, app = dbos_fastapi
+
+    @app.middleware("http")
+    async def authMiddleware(
+        request: Request, call_next: Callable[[Request], Awaitable[Response]]
+    ) -> Response:
+        with DBOSContextEnsure() as ctx:
```

Comment:
Is the ContextEnsure necessary?  What about the handler context DBOS creates in its own middleware?  Is it not working?

Reply:
Got it!  Can we make `set_authentication` a method of `DBOS` instead?  It feels awkward to ask users to access the raw context directly, which is something I think we should keep private. This is the only place where it's exposed.

Reply:
I can definitely add set_authentication, to reduce knowledge of the context... but I'm not sure it is possible or wise to keep middleware authors and other power users away from the idea of context completely... could be something event receivers need to set up also.

Reply:
Done.